### ISSUE TITLE
Rendering Error

### ISSUE THREAD

**Issue by thespacedeck** [iconify/iconify#84]
Good day, 

I've cloned and install the Express server and provided my own icons - it works great!!! I have close to 1200 icons and seems there are a few that does not render.

this is whats rendered via the server:
```
{"prefix":"tt-flags","icons":{"ke":{"body":"  <defs>    <path stroke-miterlimit=\"10\" d=\"M-28.58 47.5l1.733 1 46.713-80.91c2.732-.73 4.232-3.33 5.732-5.927 1-1.732 5-8.66 6.734-17.663-6.93 6.003-10.93 12.93-11.93 14.663-1.5 2.598-3 5.196-2.268 7.928z\" id=\"a\"/>  </defs>  <path d=\"M0 0h512v512H0z\" fill=\"#fff\"/>  <path d=\"M0 0h512v153.6H0z\"/>  <path d=\"M0 358.4h512V512H0z\" fill=\"#060\"/>  <g id=\"b\" transform=\"matrix(3.2 0 0 3.2 255.854 256)\">    <use height=\"100%\" width=\"100%\" xlink:href=\"#a\" stroke=\"#000\"/>    <use height=\"100%\" width=\"100%\" xlink:href=\"#a\" fill=\"#fff\"/>  </g>  <use height=\"100%\" width=\"100%\" xlink:href=\"#b\" transform=\"matrix(-1 0 0 1 511.708 0)\"/>  <path d=\"M255.854 102.4c-19.2 0-51.2 51.2-60.8 76.8H0v153.6h195.055c9.6 25.6 41.6 76.8 60.8 76.8 19.2 0 51.2-51.2 60.8-76.8H512V179.2H316.654c-9.6-25.6-41.6-76.8-60.8-76.8z\" fill=\"#b00\"/>  <path id=\"c\" d=\"M316.654 332.8c9.6-25.6 16-51.2 16-76.8 0-25.6-6.4-51.2-16-76.8-9.6 25.6-16 51.2-16 76.8 0 25.6 6.4 51.2 16 76.8\"/>  <use height=\"100%\" width=\"100%\" xlink:href=\"#c\" transform=\"matrix(-1 0 0 1 511.708 0)\"/>  <g transform=\"matrix(3.2 0 0 3.2 255.854 256)\" fill=\"#fff\">    <ellipse rx=\"4\" ry=\"6\"/>    <path id=\"d\" d=\"M1 5.85s4 8 4 21-4 21-4 21z\"/>    <use height=\"100%\" width=\"100%\" xlink:href=\"#d\" transform=\"scale(-1)\"/>    <use height=\"100%\" width=\"100%\" xlink:href=\"#d\" transform=\"scale(-1 1)\"/>    <use height=\"100%\" width=\"100%\" xlink:href=\"#d\" transform=\"scale(1 -1)\"/>  </g>"}},"width":512,"height":512}
```

this is whats rendered by the vue ison component:
```
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="w-6 h-6 bg-white rounded-full iconify iconify--tt-flags iconify--troop" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 512 512" style="vertical-align: -0.125em;">  <defs>    <path stroke-miterlimit="10" iconify-vue-34="M-28.58 47.5l1.733 1 46.713-80.91c2.732-.73 4.232-3.33 5.732-5.927 1-1.732 5-8.66 6.734-17.663-6.93 6.003-10.93 12.93-11.93 14.663-1.5 2.598-3 5.196-2.268 7.928z" id="iconify-vue-31"></path>  </defs>  <path iconify-vue-34="M0 0h512v512H0z" fill="#fff"></path>  <path iconify-vue-34="M0 0h512v153.6H0z"></path>  <path iconify-vue-34="M0 358.4h512V512H0z" fill="#060"></path>  <g id="iconify-vue-32" transform="matrix(3.2 0 0 3.2 255.854 256)">    <use height="100%" width="100%" xlink:href="#iconify-vue-31" stroke="#000"></use>    <use height="100%" width="100%" xlink:href="#iconify-vue-31" fill="#fff"></use>  </g>  <use height="100%" width="100%" xlink:href="#iconify-vue-32" transform="matrix(-1 0 0 1 511.708 0)"></use>  <path iconify-vue-34="M255.854 102.4c-19.2 0-51.2 51.2-60.8 76.8H0v153.6h195.055c9.6 25.6 41.6 76.8 60.8 76.8 19.2 0 51.2-51.2 60.8-76.8H512V179.2H316.654c-9.6-25.6-41.6-76.8-60.8-76.8z" fill="#b00"></path>  <path id="iconify-vue-33" iconify-vue-34="M316.654 332.8c9.6-25.6 16-51.2 16-76.8 0-25.6-6.4-51.2-16-76.8-9.6 25.6-16 51.2-16 76.8 0 25.6 6.4 51.2 16 76.8"></path>  <use height="100%" width="100%" xlink:href="#iconify-vue-33" transform="matrix(-1 0 0 1 511.708 0)"></use>  <g transform="matrix(3.2 0 0 3.2 255.854 256)" fill="#fff">    <ellipse rx="4" ry="6"></ellipse>    <path id="iconify-vue-34" iconify-vue-34="M1 5.85s4 8 4 21-4 21-4 21z"></path>    <use height="100%" width="100%" xlink:href="#iconify-vue-34" transform="scale(-1)"></use>    <use height="100%" width="100%" xlink:href="#iconify-vue-34" transform="scale(-1 1)"></use>    <use height="100%" width="100%" xlink:href="#iconify-vue-34" transform="scale(1 -1)"></use>  </g></svg>
```

This is the implementation: 
```
<Icon :icon="`@troop:tt-flags:ke`" :inline="true" class="w-6 h-6 bg-white rounded-full" />
```

Environment: 
Chrome: Version 94.0.4606.61 (Official Build) (x86_64)
"@iconify/json": "^1.1.405",
"@iconify/vue": "^3.0.0",
"vue": "^3.2.6",
"vite": "^2.6.1",
"@vitejs/plugin-vue": "^1.9.2",

I'm unsure why the component is generating the "iconify-vue-34=" instead of the normal "d="...

Your assistence is appreciated.

Kind Reagrds,
Wzz

**Comment by cyberalien** [comment]
This is very weird. I've just tested it, that icon renders correctly. Can you post more complete example of icon set? Maybe there is something else that triggers error.

**Comment by cyberalien** [comment]
I figured out why it happens, but it doesn't happen with code you've posted. In your code you are using shapes with `id="a"` and up to `id="c"`. That code works fine. But if shape has `id="d"`, it will cause that bug.

Problem is each icon should have unique ids, so component replaces ids in rendered icons. Proper id replacement would require parsing code and analysing each shape, which would require too much coding, so id replacement algorithm is very basic. It finds `id="d"` and attempts to replace all entries, but there are also entries where `d` is attribute and it incorrectly replaces it. I didn't account for possibility of ids matching attribute names.

I'll fix it. For now I suggest to use SVGO on your icons to optimise them and replace ids. Make sure you include `cleanupIDs` plugin with `prefix` set to something, like `id-`. This is example of option when running SVGO in Node.js app:
```json
{
	cleanupIDs: {
		remove: true,
		prefix: 'id-',
	},
}
```

**Comment by cyberalien** [comment]
Fixed in `@iconify/vue` version 3.1.0-beta.5. You can install it using `npm install @iconify/vue@next`.

In addition to this fix, 3.1 versions have proper ES modules support and big internal changes in handling API queries. They are stable, but before removing `next` tag I want to do more testing with using new versions in various projects.

**Comment by thespacedeck** [comment]
Incredible @cyberalien, thanks for the fix - 3.1.0-beta.5 works like a charm.
I'll monitor the releases.

**Comment by cyberalien** [comment]
Forgot to close this issue. Fix made it to stable versions of all icon components few weeks ago.
